docs/cancel_http_requests.md

```diff
@@ -32,25 +40,36 @@ class CancellableHttpBackend extends HttpBackend {
   async createRequest<T>(
     [...]
     let response;
+    
     try {
-      response = await axios.request<T>({
-        url: url + this.serialize(query),
-        method: method ?? 'GET',
-        headers: headers,
-        responseType: resType,
+      const response = await fetch(urlWithQuery, {
+        method,
+        headers,
+        body: JSON.stringify(data),
         signal: this.abortController.signal,
-        transformResponse,
-        timeout: timeout,
-        data: data,
       });
-    } catch (err: any) {
-      throw new HttpResponseError(
-        `Http error response: (${err.response.status}) ${JSON.stringify(err.response.data)}`,
-        err.response.status as STATUS_CODE,
-        err.response.statusText,
-        JSON.stringify(err.response.data),
-        url + this.serialize(query)
-      )
+
+      if (typeof response === 'undefined') {
+        throw new Error('Response is undefined');
+      }
+
+      // Handle responses with status code >= 400
+      if (response.status >= 400) {
+        const errorData = await response.text();
+        throw new HttpResponseError(
+         ...
+        );
+      }
+
+      if (json) {
+        return response.json() as T;
+      } else {
+        return response.text() as unknown as T;
+      }
+    } catch (e: unknown) {
+      ...
+    } finally {
+      clearTimeout(t);
```

Comment:
unsure i followed this example from line 52-72, is this a fetch way to do thing?

Reply:
yes, fetch does not timeout by default. If you want to timeout your fetch requests, you'll need to send an abort signal with a manual timeout method. Line 72 just cleans the timer out of the scope.

Reply:
note that this is a doc to customize our existing `createRequest` method.